kermit/src/commonTest/kotlin/co/touchlab/kermit/KermitTest.kt

```diff
@@ -13,39 +13,142 @@
 
 package co.touchlab.kermit
 
+import co.touchlab.testhelp.concurrency.ThreadOperations
 import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertTrue
 
 class KermitTest {
 
-    private val testLogWriter = TestLogWriter(loggable = Severity.Verbose)
-    private val testConfig = TestConfig(
-        minSeverity = Severity.Verbose,
-        logWriterList = listOf(testLogWriter),
-    )
+    private fun getTestConfig(logWriterList: List<LogWriter>): TestConfig {
+        return TestConfig(
+            minSeverity = Severity.Verbose,
+            logWriterList = logWriterList,
+        )
+    }
+
+    private fun getTestLogWriter(): TestLogWriter {
+        return TestLogWriter(loggable = Severity.Verbose)
+    }
+
+    @Test
+    fun testGlobal() {
+        val testLogWriter = getTestLogWriter()
+        Logger.apply {
+            setMinSeverity(Severity.Verbose)
+            setLogWriters(testLogWriter)
+        }
+        Logger.i { "Does global log?" }
+        assertEquals(testLogWriter.logs.first().message, "Does global log?")
+    }
+
+    @Test
+    fun testGlobal_Severity() {
+        val testLogWriter = getTestLogWriter()
+        Logger.apply {
+            setMinSeverity(Severity.Assert)
+            setLogWriters(testLogWriter)
+        }
+        Logger.e { "Does global log?" }
+        assertTrue(testLogWriter.logs.isEmpty())
+    }
+
+    @Test
+    fun testGlobal_MultipleLoggers_SetVararg() {
+        val firstLogWriter = getTestLogWriter()
+        val secondaryLogWriter = TestLogWriter(loggable = Severity.Verbose)
+        Logger.apply {
+            setMinSeverity(Severity.Verbose)
+            setLogWriters(firstLogWriter, secondaryLogWriter)
+        }
+        Logger.e { "Message" }
+        assertTrue(firstLogWriter.logs.size == 1)
+        assertTrue(secondaryLogWriter.logs.size == 1)
+        firstLogWriter.assertLast { message == "Message" }
+        secondaryLogWriter.assertLast { message == "Message" }
+    }
+
+    @Test
+    fun testGlobal_MultipleLoggers_SetList() {
+        val firstLogWriter = getTestLogWriter()
+        val secondaryLogWriter = TestLogWriter(loggable = Severity.Verbose)
+        Logger.apply {
+            setMinSeverity(Severity.Verbose)
+            setLogWriters(listOf(firstLogWriter, secondaryLogWriter))
+        }
+        Logger.e { "Message" }
+        assertTrue(firstLogWriter.logs.size == 1)
+        assertTrue(secondaryLogWriter.logs.size == 1)
+        firstLogWriter.assertLast { message == "Message" }
+        secondaryLogWriter.assertLast { message == "Message" }
+    }
```

Comment:
What about another version of this test where the severity is different so only one of the writers actually triggers? Or is that already covered elsewhere?

Reply:
It's covered in non-global version. I'll double check